nautobot/dcim/models/device_components.py

```diff
@@ -88,10 +87,6 @@ def to_objectchange(self, action, **kwargs):
 
         return super().to_objectchange(action, related_object=device, **kwargs)
 
-    @property
-    def parent(self):
-        return getattr(self, "device", None)
-
```

Comment:
Defining `parent` on the `ComponentModel` base class was problematic because it was shadowing the `InventoryItem.parent` DB field. This appears to have been the case for a LONG time - no idea why it didn't cause issues before.

It's really only needed on `CableTermination` subclasses, not on all `ComponentModels`, and fortunately `InventoryItem` isn't a `CableTermination`, so I was able to move it as seen below.